ui/src/user/auth-actions.ts

```diff
@@ -0,0 +1,166 @@
+                method: 'POST',
+                data: {name},
+                // eslint-disable-next-line @typescript-eslint/naming-convention
+                headers: {Authorization: 'Basic ' + Base64.encode(username + ':' + password)},
+            });
+            dispatch(
+                uiActions.addSnackMessage(`A client named '${name}' was created for your session.`)
+            );
+            localStorage.setItem(tokenKey, response.data.token);
+            return await dispatch(tryAuthenticate());
+        } catch (error) {
+            dispatch(uiActions.addSnackMessage('Login failed'));
+            return Promise.reject(new Error('Login failed.'));
+        } finally {
+            dispatch(authActions.isAuthenticating(false));
+        }
+    };
+};
+
+export const tryAuthenticate = () => {
+    return async (dispatch: AppDispatch) => {
+        if (!getAuthToken()) {
+            return Promise.reject(new Error('No token provided'));
+        }
+
+        try {
+            const response = await axios
+                .create()
+                // eslint-disable-next-line @typescript-eslint/naming-convention
+                .get<IUser>(config.get('url') + 'current/user', {
+                    headers: {'X-Gotify-Key': getAuthToken()},
+                });
+            dispatch(authActions.login(response.data));
+            dispatch(uiActions.setConnectionErrorMessage(null));
+            reconnectTime = 7500;
+        } catch (error: unknown) {
+            if (error instanceof AxiosError) {
+                if (!error || !error.response) {
+                    dispatch(connectionError('No network connection or server unavailable.'));
+                    return Promise.reject(error);
+                }
+
+                if (error.response.status >= 500) {
+                    dispatch(
+                        connectionError(
+                            `${error.response.statusText} (code: ${error.response.status}).`
+                        )
+                    );
+                    return Promise.reject(new Error(error.message));
+                }
+
+                dispatch(uiActions.setConnectionErrorMessage(null));
+
+                if (error.response.status >= 400 && error.response.status < 500) {
+                    dispatch(logout());
+                }
+            }
+            return Promise.reject(error);
+        }
+    };
+};
+
+export const logout = () => {
+    return async (dispatch: AppDispatch) => {
+        await axios
+            .get(config.get('url') + 'client')
+            .then((resp: AxiosResponse<IClient[]>) => {
+                resp.data
+                    .filter((client) => client.token === getAuthToken())
+                    .forEach((client) => axios.delete(config.get('url') + 'client/' + client.id));
+            })
+            .catch(() => Promise.resolve());
+
+        localStorage.removeItem(tokenKey);
+        dispatch(authActions.logout());
```

Comment:
This naming is weird. In a file called auth-actions, we have a function called logout which uses authActions.logout from auth-slice.ts. Is this really the redux way? Having a filed called as a variable from another file which use each other seems complicated.

Reply:
In the reducer you must not do any side effect, that is the reason, why the reducer and the action (sideeffect) is seperated, see also here:
https://redux.js.org/usage/migrating-to-modern-redux#reducers-and-actions-with-createslice

But yes, we can change the names.
Maybe use in the action doLogin and in the reducer loginDone or what naming you suggest?